### PR TITLE
FOGL-7717 - write multiple values in a single call with opcuaclient

### PR DESCRIPTION
**Pull request by ashish-jabble**

### CODE REVIEW

**Review by MarkRiddoch — APPROVED** [review]

File: python/fledge/plugins/north/opcuaclient/opcuaclient.py [hunk 1]
```diff
@@ -213,92 +219,117 @@ def __init__(self, config):
...
+                if self.server_certificate:
+                    if not self.server_certificate.endswith(valid_cert_extensions):
+                        _LOGGER.warning("Server certificate must have either in DER or PEM format.")
+                    else:
+                        cert_path = "{}pem/".format(self.certs_dir) if str(self.server_certificate).endswith(
+                            '.pem') else self.certs_dir
+                        server_certificate = "{}{}".format(cert_path, self.server_certificate)
+                if self.client_certificate:
+                    if not self.client_certificate.endswith(valid_cert_extensions):
+                        _LOGGER.warning("Client certificate must have either in DER or PEM format.")
```
Comment (RVerhoeff) [inline python/fledge/plugins/north/opcuaclient/opcuaclient.py]: Client certificate must be either in DER or PEM format

File: python/fledge/plugins/north/opcuaclient/opcuaclient.py [hunk 2]
```diff
@@ -213,92 +219,117 @@ def __init__(self, config):
...
                 else:
-                    cert_path = "{}pem/".format(self.certs_dir) if str(self.client_certificate).endswith(
-                        '.pem') else self.certs_dir
-                    certificate = "{}{}".format(cert_path, self.client_certificate)
-            else:
-                _LOGGER.warning("Client certificate cannot be empty and must have either in DER or PEM format.")
-            if self.client_private_key:
-                if not str(self.client_private_key).endswith('.pem'):
-                    _LOGGER.warning("Private key must have in PEM format.")
+                    _LOGGER.warning("Client certificate cannot be empty and must have either in DER or PEM format.")
```
Comment (RVerhoeff) [inline python/fledge/plugins/north/opcuaclient/opcuaclient.py]: "Client certificate cannot be empty and must be either in DER or PEM"

File: python/fledge/plugins/north/opcuaclient/opcuaclient.py [hunk 3]
```diff
@@ -213,92 +219,117 @@ def __init__(self, config):
...
+                if self.client_private_key:
+                    if not str(self.client_private_key).endswith('.pem'):
+                        _LOGGER.warning("Private key must have in PEM format.")
```
Comment (RVerhoeff) [inline python/fledge/plugins/north/opcuaclient/opcuaclient.py]: "Private key must be in PEM format."

File: python/fledge/plugins/north/opcuaclient/opcuaclient.py [hunk 4]
```diff
@@ -213,92 +219,117 @@ def __init__(self, config):
...
-            client.application_uri = app_uri[0]
-            await client.set_security(policy=policy, certificate=certificate, private_key=private_key,
-                                      private_key_password=passphrase, server_certificate=server_certificate, mode=mode)
-        async with client:
-            var = client.get_node(payload_block["node"])
-            user_ts = datetime.strptime(payload_block["timestamp"], '%Y-%m-%d %H:%M:%S.%f%z')
-            data_value = ua.DataValue(Value=self._value_to_variant(payload_block["value"], payload_block["type"]),
-                                      SourceTimestamp=user_ts)
-            await var.write_value(data_value)
+                    _LOGGER.warning("Private Key cannot be empty and must have in PEM format.")
```
Comment (RVerhoeff) [inline python/fledge/plugins/north/opcuaclient/opcuaclient.py]: "Private Key cannot be empty and must be in PEM format."

**Review by RVerhoeff — APPROVED** [review]
Few fixes in error messages. They are minor and can be addressed later.

**Pull request merged**